server/api/recovery.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"net/http"
+
+	"github.com/pingcap/pd/pkg/apiutil"
+	"github.com/pingcap/pd/server"
+	"github.com/unrolled/render"
+)
+
+type recoveryHandler struct {
```

Comment:
I think we should not call `recovery*` here including the filename. Supporting the recovery feature is our target to change the code this time but not for the code self.

We should name it according to what we want to supply in the view of PD.

Reply:
how about `/admin/reset_ts`?